src/logger/fileWriter.c

```diff
@@ -244,94 +260,152 @@ static int openNewLogfile(char *filename)
-    unsigned int flushTimeoutInterval = 0;
-    portTickType flushTimeoutStart = 0;
-    size_t tick = 0;
-    enum writing_status writingStatus = WRITING_INACTIVE;
-    char filename[FILENAME_LEN];
-
-    while(1) {
+        LoggerMessage *msg = NULL;
+        struct file_status fs = { 0 };
+
         while(1) {
-            //wait for the next sample record
-            xQueueReceive(g_sampleRecordQueue, &(msg), portMAX_DELAY);
+                int ok = 0;
 
-            if ((LoggerMessageType_Start == msg->type || LoggerMessageType_Sample == msg->type) &&
-                WRITING_INACTIVE == writingStatus) {
-                pr_debug("Logging: Start\r\n");
-                LED_disable(3);
-                flushTimeoutInterval = FLUSH_INTERVAL_MS;
-                flushTimeoutStart = xTaskGetTickCount();
-                tick = 0;
-                writingStatus = openNewLogfile(filename);
-            } else if (LoggerMessageType_Stop == msg->type) {
-                pr_info_int_msg("Logging: wrote ", tick);
-                break;
-            }
-
-            if (LoggerMessageType_Sample == msg->type && WRITING_ACTIVE == writingStatus) {
-
-                if (0 == tick) {
-                    writeHeaders(msg->channelSamples, msg->sampleCount);
-                }
-                LED_toggle(2);
-                int rc = writeChannelSamples(msg->channelSamples, msg->sampleCount);
-                if (rc == WRITE_FAIL) {
-                    LED_enable(3);
-                    //try to recover
-                    f_close(g_logfile);
-                    UnmountFS();
-                    pr_error("file: Error writing, recovering..\r\n");
-                    InitFS();
-                    rc = openLogfile(g_logfile, filename);
-                    if (0 != rc) {
-                        pr_error_str_msg("file: error recovering ", filename);
+                // Get a sample.
+                xQueueReceive(g_sampleRecordQueue, &(msg), portMAX_DELAY);
+
+                switch (msg->type) {
+                case LoggerMessageType_Sample:
+                        ok = logging_sample(&fs, msg);
+                        break;
+                case LoggerMessageType_Start:
+                        ok = logging_start(&fs);
+                        break;
+                case LoggerMessageType_Stop:
+                        ok = logging_stop(&fs);
                         break;
-                    } else {
-                        LED_disable(3);
-                    }
+                default:
+                        pr_warning("Unsupported message type\r\n");
+                        ok = false;
                 }
-                LED_disable(3);
-                if (isTimeoutMs(flushTimeoutStart, flushTimeoutInterval)) {
-                    flushLogfile(g_logfile);
-                    flushTimeoutStart = xTaskGetTickCount();
+
+                /* Turns the LED on if things are bad, off otherwise. */
+                file_led_set(ok != 0);
```

Comment:
Well... I left this as a logging_led because that is what we are using it for.  File led is ambiguous if we have more than one file in operation.